kubeflow/pipelines.py

```diff
@@ -0,0 +1,445 @@
+        checkpoint_file (str): Path to an estimator checkpoint file.
+        test_data (str): Path to test dataset directory.
+        tb_log_dir (str): Path to tensorbload log directory.
+        checkpoint_dir (str): Path to checkpoint file directory.
+        volume (kfp.dsl.PipelineVolume): The volume where datasets are stored.
+        memory_limit (str): Set memory limit for this operator. For simplicity,
+            we set memory_request = memory_limit.
+        num_gpu (int): Set the number of GPU for this operator
+        gpu_type (str): Set the type of GPU
+
+    Returns:
+        kfp.dsl.ContainerOp: Represents an op implemented by a container image
+            to evaluate an estimator.
+    """
+    evaluate = dsl.ContainerOp(
+        name="evaluate",
+        image=docker,
+        command=["datasetinsights", "evaluate"],
+        arguments=[
+            f"--config={config}",
+            f"--checkpoint-file={checkpoint_file}",
+            f"--test-data={test_data}",
+            f"--tb_log_dir={tb_log_dir}",
+        ],
+        pvolumes={DATA_PATH: volume},
+    )
+    # GPU
+    evaluate.set_gpu_limit(num_gpu)
+    evaluate.add_node_selector_constraint(
+        "cloud.google.com/gke-accelerator", gpu_type
+    )
+
+    evaluate.set_memory_request(memory_limit)
+    evaluate.set_memory_limit(memory_limit)
+
+    evaluate.apply(gcp.use_gcp_secret("user-gcp-sa"))
+
+    return evaluate
+
+
+@dsl.pipeline(
+    name="Train on the SynthDet sample",
+    description="Train on the SynthDet sample",
+)
+def train_on_synthdet_sample(
+    docker: str = "unitytechnologies/datasetinsights:latest",
+    source_uri: str = (
+        "https://storage.googleapis.com/datasetinsights/data/"
+        "synthetic/SynthDet.zip"
+    ),
+    config: str = "datasetinsights/configs/faster_rcnn_synthetic.yaml",
+    tb_log_dir: str = "gs://<bucket>/runs/yyyymmdd-hhmm",
+    checkpoint_dir: str = "gs://<bucket>/checkpoints/yyyymmdd-hhmm",
+    volume_size: str = "100Gi",
+):
+    output = train_data = val_data = DATA_PATH
+
+    # The following parameters can't be `PipelineParam` due to this issue:
+    # https://github.com/kubeflow/pipelines/issues/1956
+    # Instead, they have to be configured when the pipeline is compiled.
+    memory_limit = "64Gi"
+    num_gpu = 8
+    gpu_type = "nvidia-tesla-v100"
+
+    # Pipeline definition
+    vop = volume_op(volume_size=volume_size)
+    download = download_op(
+        docker=docker,
+        source_uri=source_uri,
+        output=output,
+        volume=vop.volume,
+        memory_limit=memory_limit,
+    )
+    train = train_op(
+        docker=docker,
+        config=config,
+        train_data=train_data,
+        val_data=val_data,
+        tb_log_dir=tb_log_dir,
+        checkpoint_dir=checkpoint_dir,
+        volume=download.pvolumes[DATA_PATH],
+        memory_limit=memory_limit,
+        num_gpu=num_gpu,
+        gpu_type=gpu_type,
+    )
+
+    return train
```

Comment:
What is the purpose of the return calls here and in the other dsl.pipelines?

Reply:
It's not used in these pipelines. It simply prevent `flake8` error as train variable is defined but not used. I will remove these return.